### PR TITLE
Add matching_strategy: str = "all" to the SearchParameters model on f…

### PR DESCRIPTION
**Pull request by viktoriussuwandi**
Hi [sanders41](https://github.com/sanders41),
just trying to help on [issue#433](https://github.com/sanders41/meilisearch-fastapi/issues/433), please check it out

### CODE REVIEW

File: meilisearch_fastapi/models/search_parameters.py [hunk 1]
```diff
@@ -19,3 +19,6 @@ class SearchParameters(CamelBase):
     highlight_pre_tag: str = "<em>"
     highlight_post_tag: str = "</em>"
     crop_marker: str = "..."
+    matching_strategy: str = "all"
+    hists_per_page: int | None = None
+    page: int | None = None
```
Comment (sanders41) [inline meilisearch_fastapi/models/search_parameters.py:24]: You did this correctly according to the issue, but I am realizing looking at it now that Pydantic is not going to like this in Python 3.8 so can you make the following update?

```suggestion
    hits_per_page: Optional[int] = None
    page: Optional[int] = None
```

**Review by sanders41 — CHANGES_REQUESTED** [review]
Thanks @viktoriussuwandi!

Can you also add a test for this in [test_search_routes.py](https://github.com/sanders41/meilisearch-fastapi/blob/main/tests/test_search_routes.py)? If you have any questions about how to do that let me know.

**Comment by sanders41** [comment]
Making the requested change will fix the failing tests.

**Comment by viktoriussuwandi** [comment]
> Thanks @viktoriussuwandi!
> 
> Can you also add a test for this in [test_search_routes.py](https://github.com/sanders41/meilisearch-fastapi/blob/main/tests/test_search_routes.py)? If you have any questions about how to do that let me know.

I am very happy to work on this.
But, since I have no idea on this test section, and avoid any mistakes happen.
If you don't mind, would you let me know step by step how to do it

Appeciate your guidance @sanders41

**Comment by sanders41** [comment]
Lets add two tests, one for `all` and one for `last`:

```py
async def test_custom_search_params_with_matching_strategy_all(test_client, index_with_documents):
    uid, _ = index_with_documents
    data = {
        "uid": uid,
        "query": "man loves",
        "limit": 5,
        "matchingStrategy": "all",
    }

    response = await test_client.post("/search", json=data)
    assert len(response.json()["hits"]) == 1
```

and

```py
async def test_custom_search_params_with_matching_strategy_last(test_client, index_with_documents):
    uid, _ = index_with_documents
    data = {
        "uid": uid,
        "query": "man loves",
        "limit": 5,
        "matchingStrategy": "last",
    }

    response = await test_client.post("/search", json=data)
    assert len(response.json()["hits"]) > 1
```

In coming up with tests I also came across another issue. In `search_routes.py` we also need to add the new values into the search. So the `return` for `async def search` needs to be updated to:

```py
    return await index.search(
        query=search_parameters.query,
        offset=search_parameters.offset,
        limit=search_parameters.limit,
        filter=search_parameters.filter,
        facets=search_parameters.facets,
        attributes_to_retrieve=search_parameters.attributes_to_retrieve,
        attributes_to_crop=search_parameters.attributes_to_crop,
        sort=search_parameters.sort,
        crop_length=search_parameters.crop_length,
        attributes_to_highlight=search_parameters.attributes_to_highlight,
        show_matches_position=search_parameters.show_matches_position,
        highlight_pre_tag=search_parameters.highlight_pre_tag,
        highlight_post_tag=search_parameters.highlight_post_tag,
        crop_marker=search_parameters.crop_marker,
        matching_strategy=search_parameters.matching_strategy,
        hits_per_page=search_parameters.hits_per_page,
        page=search_parameters.page,
    )
```

**Review by sanders41 — APPROVED** [review]
LGTM!

**Pull request merged**